frontend/src/react-query/api/schema.tsx

```diff
@@ -0,0 +1,469 @@
+  SchemaRegistrySubjectDetails,
+  SchemaVersion,
+} from 'state/restInterfaces';
+import { formatToastErrorMessageGRPC } from 'utils/toast.utils';
+
+// Stale time constants for consistent cache behavior
+// const STALE_TIME_SHORT = 10_000; // 10 seconds
+const STALE_TIME_MEDIUM = 30_000; // 30 seconds
+// const STALE_TIME_LONG = 60_000; // 60 seconds
+
+export const useListSchemasQuery = () => {
+  return useQuery<SchemaRegistrySubject[]>({
+    queryKey: ['schemaRegistry', 'subjects'],
+    queryFn: async () => {
+      const response = await fetch(`${config.restBasePath}/schema-registry/subjects`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${config.jwt}`,
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch schema subjects');
+      }
+
+      const data = await response.json();
+      // could also be a "not configured" response
+      if (!Array.isArray(data)) {
+        return [];
+      }
+
+      return data;
+    },
+    refetchOnMount: true,
+  });
+};
+
+export const useSchemaModeQuery = () =>
+  useQuery<string | null>({
+    queryKey: ['schemaRegistry', 'mode'],
+    queryFn: async () => {
+      const response = await fetch(`${config.restBasePath}/schema-registry/mode`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${config.jwt}`,
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch schema mode');
+      }
+
+      const data = await response.json();
+      if (data.isConfigured === false) {
+        return null;
+      }
+
+      return data.mode ?? null;
+    },
+    refetchOnMount: true,
+  });
+
+export const useSchemaCompatibilityQuery = () =>
+  useQuery<string | null>({
+    queryKey: ['schemaRegistry', 'compatibility'],
+    queryFn: async () => {
+      const response = await fetch(`${config.restBasePath}/schema-registry/config`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${config.jwt}`,
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to fetch schema compatibility');
+      }
+
+      const data = await response.json();
+      if (data.isConfigured === false) {
+        return null;
+      }
+
+      return data.compatibility ?? null;
+    },
+    refetchOnMount: true,
+  });
+
+export const useSchemaDetailsQuery = (subjectName: string, options?: { enabled?: boolean }) =>
+  useQuery<SchemaRegistrySubjectDetails>({
+    queryKey: ['schemaRegistry', 'subjects', subjectName, 'details'],
+    queryFn: async () => {
+      const response = await fetch(
+        `${config.restBasePath}/schema-registry/subjects/${encodeURIComponent(subjectName)}/versions/all`,
+        {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${config.jwt}`,
+          },
+        }
+      );
```

Comment:
If it's getting repetitive, let's create `authenticatedFetch` with 
```tsx
          headers: {
            Authorization: `Bearer ${config.jwt}`,
          },
```
already added